File: design-pattern-java/app/src/test/java/design/pattern/java/patterns/behavioral/command/CommandTest.java
package design.pattern.java.patterns.behavioral.command;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class CommandTest {

  class ServiceMock extends Service {
    private Boolean wasCalled = false;

    @Override
    public void apply(String value) {
      this.wasCalled = true;
      super.apply(value);
    }

    public Boolean hasBeenCalled() {
      return this.wasCalled;
    }
  }

  @Test
  public void shouldCallMethodWhenCommandExecutes() {
    ServiceMock testSrv = new ServiceMock();
    ServiceCommandBuilder testCommand = new ServiceCommandBuilder(testSrv);
    testCommand.setArgs("value").Run();
    Assertions.assertTrue(testSrv.hasBeenCalled());
  }

}

File: design-pattern-jvm/app/src/test/kotlin/design/pattern/kotlin/patterns/behavioral/command/CommandTest.kt
package design.pattern.kotlin.patterns.behavioral.command

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class CommandTestKt {
    internal inner class ServiceMock : Service() {
        private var wasCalled = false
        override fun apply(value: String?) {
            wasCalled = true
            super.apply(value)
        }

        fun hasBeenCalled(): Boolean {
            return wasCalled
        }
    }

    @Test
    fun shouldCallMethodWhenCommandExecutes() {
        val testSrv: ServiceMock = ServiceMock()
        val testCommand = ServiceCommandBuilder(testSrv)
        testCommand.setArgs("value").Run()
        Assertions.assertTrue(testSrv.hasBeenCalled())
    }
}
